demisto_sdk/commands/common/hook_validations/integration.py

```diff
@@ -81,6 +89,7 @@ def core_integration_validations(self, validate_rn: bool = True):
         """
         answers = [
             super().is_valid_file(validate_rn),
+            super().validate_readme_exists("integration", self.is_modified, self.is_added, self.validate_all),
```

Comment:
Instead of passing the type of the class as an argument, It is better to access the child's class name from the parent function (see [here](https://stackoverflow.com/questions/43745055/child-class-name-in-parent-class-function-at-run-time-python)). Also, you don't have to pass the variables to the parent class since it can access the child's variables (see [here](https://stackoverflow.com/questions/49436367/access-child-class-variable-in-parent-class)). Same for all the others.

Reply:
i think there are some problems with this approch i will talk to you about it